src/content/docs/zh-cn/develop/sidecar.mdx

```diff
@@ -0,0 +1,199 @@
+  extension = '.exe';
+}
+
+async function main() {
+  const rustInfo = (await execa('rustc', ['-vV'])).stdout;
+  const targetTriple = /host: (\S+)/g.exec(rustInfo)[1];
+  if (!targetTriple) {
+    console.error('Failed to determine platform target triple');
+  }
+  fs.renameSync(
+    `src-tauri/binaries/sidecar${extension}`,
+    `src-tauri/binaries/sidecar-${targetTriple}${extension}`
+  );
+}
+
+main().catch((e) => {
+  throw e;
+});
+```
+
+注意，如果编译的体系结构与运行的体系结构不同，这个脚本将无法工作。
+
+## 从 Rust 运行
+
+在 Rust 端, 导入 `tauri_plugin_shell::ShellExt` 特征 并在AppHandle上调用 `shell().sidecar()` 函数。
+
+```rust
+use tauri_plugin_shell::ShellExt;
+use tauri_plugin_shell::process::CommandEvent;
+
+// `sidecar()` 只需要文件名, 不像 JavaScript 中的整个路径
+let sidecar_command = app.shell().sidecar("my-sidecar").unwrap();
+let (mut rx, mut _child) = sidecar_command
+  .spawn()
+  .expect("Failed to spawn sidecar");
+
+tauri::async_runtime::spawn(async move {
+  // 读取诸如stdout之类的事件
+  while let Some(event) = rx.recv().await {
+    if let CommandEvent::Stdout(line) = event {
+      window
+        .emit("message", Some(format!("'{}'", line)))
+        .expect("failed to emit event");
+      // 写入标准输入
+      child.write("message from Rust\n".as_bytes()).unwrap();
+    }
+  }
+});
+```
+
+您可以将此代码放在 Tauri 命令中，以便轻松传递 AppHandle，或者您可以在构建脚本中存储对 AppHandle 的引用，以便在应用程序的其他位置访问它。
+
+## 从 JavaScript 运行
+
+在 JavaScript 代码中，从 `@tauri-apps/plugin-shell` 模块中导入 `Command` 类，并使用其静态方法 `sidecar`。
+
+```javascript
+import { Command } from '@tauri-apps/plugin-shell';
+// `binaries/my-sidecar` 是在 `tauri.conf.json > tauri > bundle > externalBin` 指定的确切值。
+const sidecar_command = Command.sidecar('binaries/my-sidecar');
+const output = await sidecar_command.execute();
+```
+
+## 传递参数
+
+你可以像运行普通的 `Command` 一样向 Sidecar 命令传递参数。
+
+参数可以是**静态的**（例如 `-o` 或 `serve`）或者**动态的**（例如 `<file_path>` 或者 `localhost:<PORT>`）。
+你按照调用它们的确切顺序来定义这些参数。静态参数保持不变，而动态参数可以使用正则表达式来定义。
+
+首先，在 `src-tauri/capabilities/main.json` 中定义需要传递给 sidecar 命令的参数。
+
+```json title="src-tauri/capabilities/main.json" ins={14-31}
+{
+  "$schema": "../gen/schemas/desktop-schema.json",
+  "identifier": "default",
+  "description": "Capability for the main window",
+  "windows": ["main"],
+  "permissions": [
+    "path:default",
+    "event:default",
+    "window:default",
+    "app:default",
+    "resources:default",
+    "menu:default",
+    "tray:default",
+    {
+      "identifier": "shell:allow-execute",
+      "allow": [
+        {
+          "args": [
+            "arg1",
+            "-a",
+            "--arg2",
+            {
+              "validator": "\\S+"
+            }
+          ],
+          "cmd": "",
+          "name": "binaries/my-sidecar",
+          "sidecar": true
+        }
+      ]
+    },
+    "shell:allow-open"
+  ]
+}
+```
+
+:::note
+如果你正在从 Tauri v1 迁移，Tauri v2 CLI 中的 `migrate` 命令应该会为你处理这个问题。阅读[自动迁移](/start/migrate/from-tauri-1/#automated-migration)了解更多信息。
+:::
+
+然后，要调用 sidecar 命令，只需将**所有**参数作为数组传入即可。
```

Comment:
```suggestion
然后，要调用 `sidecar` 命令，只需将**所有**参数作为数组传入即可。
```

Missing backquote.

Reply:
The original text has no backquote.
![image](https://github.com/user-attachments/assets/4952e7ce-01e3-44a0-9f90-57450f0f520b)

Reply:
Does this need to be consistent with the original text? @Embers-of-the-Fire

Reply:
I think it's okay with or without, let's be in sync for now and then change it when the source material updates.